orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -149,9 +149,17 @@ class CompleteStageHandler(
         } ?: id
       }
 
+    // If startTime was not set, then assume this was instantaneous.
+    // In practice 0 startTimes can happen if there is
+    // an exception in StartStageHandler or guard causing skipping.
+    // Without a startTime, we cannot record a meaningful time,
```

Comment:
Would it make sense to just skip recording the metric entirely if we don't have a start time, rather than record a placeholder value of 0?

Reply:
There's still value in the counts. If the time were 0 because of some other error condition, the final status would be TERMINAL (or maybe NOT_STARTED if it were some non-error reason). Otherwise you would want another counter indicating errors in this counter to raise awareness to a situation where stages were not executing. That's even more complicated.

Reply:
Yeah, I guess it does provide visibility into the counts without the complexity of another metric, makes sense.